rfd/9999-aws-api-integration-using-oidc.md

```diff
@@ -0,0 +1,554 @@
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Actions": "[
+				rds:DescribeDBClusters",
+				rds:DescribeDBInstances",
+	        ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+This AWS Role must trust Teleport as an IdP.
+To do so, the user must add the following Trusted relationship:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::0123456789012:oidc-provider/proxy.example.com"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "proxy.example.com:aud": "sts.amazonaws.com"
+                }
+            }
+        }
+    ]
+}
+```
+
+### UX
+Besides the flow describe above (Users Point Of View), the Integration resource will have the usual CRUD operations via both API and `tctl`.
+The only updatable field is `aws_role_arn`.
+
+HTTP API:
+```
+Methods:
+GET .../integration
+GET .../integration/:id
+POST .../integration
+PUT .../integration/:id
+DELETE .../integration/:id
+```
+JSON representation:
+```json
+{
+	"name": "myaws",
+	"subkind": "aws-oidc"
+	"subkindData": {
+		"aws_role_arn": "arn:aws:123:TeleportOIDC"
+	}
+}
+```
+
+`tctl`:
+
+```
+$ tctl get integrations
+kind: integration
+subkind: aws-oidc
+version: v1
+metadata:
+	name: some-name
+spec:
+	aws_role_arn: arn:aws:123:TeleportOIDC
+
+$ tctl get integration/myaws
+kind: integration
+subkind: aws-oidc
+version: v1
+metadata:
+	name: some-name
+spec:
+	aws_role_arn: arn:aws:123:TeleportOIDC
+
+$ tctl create aws-integration.yaml
+```
+
+Resource representation as `yaml`:
+```yaml
+kind: integration
+subkind: aws-oidc
+version: v1
+metadata:
+	name: myaws
+spec:
+	aws_role_arn: arn:aws:123:TeleportOIDC
+```
+
+### Security
+
+#### Rotation
+JWT Signing keys can be rotated using `tctl auth rotate --type=oidc`.
+
+JWKS endpoint returns a list of public keys, so the old and the new key are provided for a seamless migration.
+
+#### MITM between AWS and Teleport
+
+The only configuration AWS has about the provider is its URL.
+If DNS is controlled by an attacker, AWS might accept requests from them.
+
+However, assuming the DNS is controlled by an attacker, then the users that are logging in to Teleport are also providing their credentials to that part.
+
+We don't think this is a scenario we should focus on this RFD.
+
+#### AWS Calls from unauthorized Teleport Users
+We'll be able to call `DescribeDBInstances` AWS endpoint, however we must ensure that this is protected by RBAC.
+
+In order to do so, we'll re-use the `db` resource with the `list` verb to allow listing RDS Databases.
+
+```yaml
+kind: role
+metadata:
+  name: editor
+spec:
+  allow:
+    rules:
+    - resources:
+      - db
+      verbs:
+      - list
+```
```

Comment:
I still don't get this tbh. In your comment you said:

> We'll have an HTTP endpoint in Proxy that calls the AWS API Describe DB Instance.

What endpoint is this and why would it be unauthenticated?

Reply:
There's still authentication, however my concern is authorization.

Let's say we have a UserA that has `editor` role and is able to List AWS RDS DBs.

However, there's also UserB that has `access` role.
Should UserB be allowed to call the List AWS RDS DBs?

Reply:
I added some more details on Listing AWS RDS DBs
https://github.com/gravitational/teleport/pull/22556/commits/83c735ba94cf63b03ad589b2f0e617e0fcda9068

Let me know if that helps in understanding my concern